adagrams/game.py

```diff
@@ -1,11 +1,169 @@
+import random
+import copy
+
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_VALUE = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+
 def draw_letters():
-    pass
+    alphabet = list(LETTER_POOL.keys())
+    drawn_letters = []
+
+    while True:
+        # Ensure return list does not have more than 10 elems
+        if len(drawn_letters) == 10:
+            break
+
+        random_letter = alphabet[random.randint(0, len(alphabet)-1)]
+        quantity = LETTER_POOL[random_letter]
+        counter = 0
+        
+        # Letter should not be in return list more than its quantity in the pool
+        for i in range(len(drawn_letters)):
+            if drawn_letters[i] == random_letter:
+                counter += 1
+        if counter < quantity:
+            drawn_letters.append(random_letter)
+
+    return drawn_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    available_letters = copy.deepcopy(letter_bank)
+    word = word.upper()
+
+    # Remove letters from available letters as they are used
+    for letter in word:
+        if letter in available_letters:
+            available_letters.remove(letter)
+        
+        # If the letter isn't in available letters, the word is invalid
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    word = word.upper()
+    score = 0
+
+    # Retrieve points for each letter
+    for letter in word:
+        score += LETTER_VALUE[letter]
+    
+    # Check for word length bonus 
+    # (words longer than 6 letters get an additional 8 points)
+    if len(word) > 6 and len(word) < 11:
```

Comment:
Do we need to check that it's also under 11 characters? In what scenarios could that happen in this code? I think it's possible that in a future version of this project, checking for upper limits is necessary, but right now, if I delete that code (so it's only `if len(word) > 6`), the tests still pass :)